### PR TITLE
lido-linea-syncswap

### PR DESCRIPTION
**Pull request by ppclunghe**
Adding Lido wstETH pools on Syncswap Linea

### CODE REVIEW

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    schema='lido_liquidity_linea',
+    alias = 'syncswap_pools',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['pool', 'time'],
+    post_hook='{{ expose_spells(\'["linea"]\',
+                                "project",
+                                "lido_liquidity",
+                                \'["pipistrella"]\') }}'
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:11]: We may use the new form of `post_hook` here:
```suggestion
    post_hook='{{ expose_spells(blockchains  = \'["linea"]\',
                                spell_type   = "project",
                                spell_name   = "lido_liquidity",
                                contributors = \'["pipistrella"]\') }}'
```

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+      t.token AS token,
+      symbol,
+      decimals,
+      AVG(price) AS price
+FROM {{source('prices','usd')}} p
+  JOIN tokens t on p.contract_address = t.token_eth
+{% if not is_incremental() %}
+WHERE DATE_TRUNC('day', p.minute) >= DATE '{{ project_start_date }}'
+{% else %}
+WHERE DATE_TRUNC('day', p.minute) >= DATE_TRUNC('day', NOW() - INTERVAL '1' day)
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:59]: We may use the `incremental_predicate` for all incremental filters:
```suggestion
WHERE {{incremental_predicate('p.minute')}}
```

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        DATE_TRUNC('hour', minute) time, 
+        0xB5beDd42000b71FddE22D3eE8a79Bd49A568fC8F as token,
+        symbol,
+        decimals,
+        last_value(price) over (partition by DATE_TRUNC('hour', minute), contract_address ORDER BY  minute range between unbounded preceding AND unbounded following) AS price
+    FROM {{ source('prices', 'usd') }} p
+    {% if not is_incremental() %}
+    WHERE DATE_TRUNC('day', p.minute) >= DATE '{{ project_start_date }}'
+    {% else %}
+    WHERE DATE_TRUNC('day', p.minute) >= DATE_TRUNC('day', NOW() - INTERVAL '1' day)
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:91]: Same as above.

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+      cr.token1,
+      SUM(CAST(amount0 AS DOUBLE)) AS amount0,
+      SUM(CAST(amount1 AS DOUBLE)) AS amount1
+ from {{source('syncswap_linea','SyncSwapClassicPool_evt_Mint')}} m
+ left join {{source('syncswap_linea','SyncSwapClassicPoolFactory_evt_PoolCreated')}} cr on m.contract_address = cr.pool 
+ 
+ {% if not is_incremental() %}
+ WHERE DATE_TRUNC('day', m.evt_block_time) >= DATE '{{ project_start_date }}'
+ {% else %}
+ WHERE DATE_TRUNC('day', m.evt_block_time) >= DATE_TRUNC('day', NOW() - INTERVAL '1' day)
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:118]: Same as above.

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+      LEFT JOIN tokens_prices_daily AS p0 ON l.time = p0.time
+      AND l.token0 = p0.token
+      LEFT JOIN tokens_prices_daily AS p1 ON l.time = p1.time
+      AND l.token1 = p1.token
+      LEFT JOIN trading_volume AS tv ON l.time = tv.time
+      AND l.pool = tv.pool
+ )
+ 
+
+select  CONCAT(CONCAT(CONCAT(CONCAT(CONCAT(blockchain,CONCAT(' ', project)) ,' '), coalesce(paired_token_symbol,'unknown')),':') , main_token_symbol, ' ', pool_type, ' ', format('%,.3f',round(coalesce(fee,0),4))) as pool_name,
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:276]: This line may be too complicated, we may simplify string concat using `||`:
```suggestion
SELECT 
    blockchain || ' ' || project || ' ' || COALESCE(paired_token_symbol, 'unknown') || ':' || main_token_symbol || ' ' || pool_type || ' ' || FORMAT('%,.3f', ROUND(COALESCE(fee, 0), 4)) AS pool_name,
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql (hunk 4 above)
Comment (ppclunghe) [reply models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql]: Thanks a lot for your suggestions, @Hosuke , I have fixed it accordingly.

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+      cr.token1,
+      SUM(CAST(amount0 AS DOUBLE)) AS amount0,
+      SUM(CAST(amount1 AS DOUBLE)) AS amount1
+ from {{source('syncswap_linea','SyncSwapClassicPool_evt_Mint')}} m
+ left join {{source('syncswap_linea','SyncSwapClassicPoolFactory_evt_PoolCreated')}} cr on m.contract_address = cr.pool 
+ 
+ {% if not is_incremental() %}
+ WHERE DATE_TRUNC('day', m.evt_block_time) >= DATE '{{ project_start_date }}'
+ {% else %}
+  WHERE {{incremental_predicate('p.minute')}}
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:118]:
```suggestion
  WHERE {{incremental_predicate('m.evt_block_time')}}
```

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+      cr.token1,
+      (-1)*SUM(CAST(amount0 AS DOUBLE)) AS amount0,
+      (-1)*SUM(CAST(amount1 AS DOUBLE)) AS amount1
+ from {{source('syncswap_linea','SyncSwapClassicPool_evt_Burn')}} b
+ left join {{source('syncswap_linea','SyncSwapClassicPoolFactory_evt_PoolCreated')}} cr on b.contract_address = cr.pool 
+ 
+ {% if not is_incremental() %}
+ WHERE DATE_TRUNC('day', b.evt_block_time) >= DATE '{{ project_start_date }}'
+ {% else %}
+ WHERE DATE_TRUNC('day', b.evt_block_time) >= DATE_TRUNC('day', NOW() - INTERVAL '1' day)
```
Comment (Hosuke) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:138]:
```suggestion
 WHERE {{incremental_predicate('b.evt_block_time')}}
```

Comment (ppclunghe) [reply models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql]: fixed

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql (hunk 6 above)
Comment (ppclunghe) [reply models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql]: yeah, fixed it

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ppclunghe 

[Sample lido_liquidity_linea_syncswap_pools](https://dune.com/queries/3683428)

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+      AVG(price) AS price
+FROM {{source('prices','usd')}} p
+  JOIN tokens t on p.contract_address = t.token_eth
+{% if not is_incremental() %}
+WHERE DATE_TRUNC('day', p.minute) >= DATE '{{ project_start_date }}'
+{% else %}
+WHERE {{incremental_predicate('p.minute')}}
+{% endif %}
+     and date_trunc('day', minute) < current_date
+     and blockchain = 'ethereum'
```
Comment (jeff-dude) [inline models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql:62]: should this be linea? or do you want to pull from L1?

**Review by jeff-dude — COMMENTED** [review]
in addition to below question, if we do flip prices filter to `linea`, do you want to wait to merge this until we see the prices data for `linea` flow into `prices.usd`?

since that was just merged, it takes a bit for the historical data to load.

File: models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql (hunk 8 above)
Comment (ppclunghe) [reply models/lido/liquidity/linea/lido_liquidity_linea_syncswap_pools.sql]: Have done with Ethereum prices only to have these pools in spell asap, 
changed for Linea prices and it works perfectly now  

Thanks a lot!

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**